includes/Main.php

```diff
@@ -1265,16 +1265,41 @@ public function set_phpmailer_validator( $mailArray ) {
 	 *
 	 * @param object $document
 	 * @param string $trigger
+	 *
 	 * @return void
 	 */
-	public function log_document_creation_to_order_notes( $document, $trigger ) {
+	public function log_document_creation_to_order_notes( object $document, string $trigger ) {
+		if ( empty( $document ) || empty( $trigger ) || ! isset( WPO_WCPDF()->settings->debug_settings['log_to_order_notes'] ) ) {
+			return;
+		}
+
 		$triggers = $this->get_document_triggers();
-		if ( ! empty( $document ) && isset( WPO_WCPDF()->settings->debug_settings['log_to_order_notes'] ) && ! empty( $trigger ) && array_key_exists( $trigger, $triggers ) ) {
-			/* translators: 1. document title, 2. creation trigger */
-			$message = __( 'PDF %1$s created via %2$s.', 'woocommerce-pdf-invoices-packing-slips' );
-			$note    = sprintf( $message, $document->get_title(), $triggers[$trigger] );
-			$this->log_to_order_notes( $note, $document );
+
+		if ( ! array_key_exists( $trigger, $triggers ) ) {
+			return;
+		}
+
+		$user_note_message = '';
+		$manual_triggers   = $this->get_document_triggers( true );
+
+		// Add user information if the trigger is manual.
+		if ( array_key_exists( $trigger, $manual_triggers ) ) {
+			$user = wp_get_current_user();
+
+			if ( ! empty( $user->display_name ) && ! empty( $user->ID )   ) {
+				$user_note_message = sprintf(
+					/* translators: 1. user display name, 2. user ID */
+					__( ' by %1$s(#%2$s)', 'woocommerce-pdf-invoices-packing-slips' ),
+					$user->display_name,
+					$user->ID
+				);
+			}
 		}
+
+		/* translators: 1. document title, 2. creation trigger, 3. user info */
+		$message = __( 'PDF %1$s created via %2$s%3$s.', 'woocommerce-pdf-invoices-packing-slips' );
```

Comment:
Could you consider removing the last placeholder? It seems unnecessary and might not aid in understanding the translation—you could concatenate it at the end instead.

Reply:
The issue is the period at the end of the sentence that made me put that placeholder, but I made some changes in the structure.